simtools/tests/test_workflow_description.py

```diff
@@ -3,82 +3,97 @@
-        == "South-MST-FlashCam-D")
-    assert (
-        file_writer._read_instrument_name(user_meta_1['PRODUCT']['ASSOCIATION'][1])
-        == "North-MST-NectarCam-7")
+    with pytest.raises(KeyError):
+        workflow_1._fill_product_association_identifier()
+
+
+def test_read_instrument_name():
 
-    user_meta_2 = get_generic_user_meta()
-    user_meta_2["PRODUCT"]["ASSOCIATION"][0]["TYPE"] = "Structure"
-    file_writer._user_meta = user_meta_2
-    file_writer._fill_user_meta()
+    _workflow = workflow.WorkflowDescription()
+    _workflow.toplevel_meta = get_generic_toplevel_meta()
 
+    _association_1 = \
+        get_generic_user_meta()['PRODUCT']['ASSOCIATION'][0]
+    _association_2 = \
+        get_generic_user_meta()['PRODUCT']['ASSOCIATION'][1]
+
+    assert (
+        _workflow._read_instrument_name(_association_1)
+        == 'South-MST-FlashCam-D')
     assert (
-        file_writer._read_instrument_name(user_meta_2["PRODUCT"]["ASSOCIATION"][0])
-        == "South-MST-Structure-D")
+        _workflow._read_instrument_name(_association_2)
+        == 'North-MST-NectarCam-7')
 
-    user_meta_3 = get_generic_user_meta()
-    user_meta_3["PRODUCT"]["ASSOCIATION"][0]["SITE"] = "Neptun"
-    file_writer._user_meta = user_meta_3
-    file_writer._fill_user_meta()
+    _association_3 = \
+        get_generic_user_meta()['PRODUCT']['ASSOCIATION'][0]
+    _association_3['SITE'] = 'Moon'
 
     with pytest.raises(ValueError):
-        file_writer._read_instrument_name(user_meta_3["PRODUCT"]["ASSOCIATION"][0])
+        _workflow._read_instrument_name(_association_3)
 
 
-def test_fill_user_meta():
+def test_merge_config_dicts():
 
-    user_meta_1 = get_generic_user_meta()
+    d_low_priority = {
+        'REFERENCE': {'VERSION': '0.1.0'},
+        'ACTIVITY': {
+            'NAME': 'SetParameterFromExternal',
+            'DESCRIPTION': 'Set data columns'
+        },
+        'DATAMODEL': 'model-A',
+        'PRODUCT': None
+    }
 
-    file_writer = writer.ModelData(
-        get_generic_workflow_config(),
-        get_generic_toplevel_meta())
-    file_writer._user_meta = user_meta_1
-    file_writer._fill_user_meta()
+    d_high_priority = {
+        'REFERENCE': {'VERSION': '0.2.0'},
+        'ACTIVITY': {'NAME': None},
+        'PRODUCT': {'DIRECTORY': './'},
+        'DATAMODEL': 'model-B'
+    }
+
+    _workflow = workflow.WorkflowDescription()
+    _workflow._merge_config_dicts(d_low_priority, d_high_priority)
 
-    user_meta_2 = {
-        'CONTACT': 'my_name'
+    d_merged = {
+        'REFERENCE': {'VERSION': '0.2.0'},
+        'ACTIVITY': {
+            'NAME': 'SetParameterFromExternal',
+            'DESCRIPTION': 'Set data columns'
+        },
+        'PRODUCT': {'DIRECTORY': './'},
+        'DATAMODEL': 'model-B',
     }
-    file_writer._user_meta = user_meta_2
 
-    with pytest.raises(KeyError):
-        file_writer._fill_user_meta()
+    assert d_merged == d_high_priority
```

Comment:
But this is exactly the intention: do modify the high-priority dict accordingly. 

Where do I see the changes you have done?

Reply:
I have not made changes, just commented in the other file. 

If you prefer to modify one of the dicts, that's fine, I don't feel that strongly about it so I resolve this coment.

Reply:
Can you point me to the comments?

Reply:
It's a comment you addressed in line 506 of [workflow_description.py](https://github.com/gammasim/gammasim-tools/pull/249/files/76fcc7cdeabbfb99044da5bf960cf55e5c22f454#diff-27012bb550b14b28ae6d7664bb52dbbc83eeca4defd62161e07bdc2b9600abfa). 

In case the direct link does not work, I put a screenshot below.
![image](https://user-images.githubusercontent.com/22728856/172918345-a36463d0-2ef9-4e90-8fbb-de11b5353ecd.png)